### PR TITLE
Update the woo widget stock links to new analytics page

### LINKED ISSUE #28570: WooCommerce Status Widget: Revise Stock Rows Links
The Start/Manage team is going to be looking at updating some of the WooCommerce wp-admin dashboard widgets, and this is the first of a few issues that Isotope will be looking into in the new year ( or possibly sooner! ).

Dashboard ‹ bend outdoors — WordPress 2020-12-11 17-03-58

This task involves updating the Stock Row link targets to point at the Analytics Stock report:

- [ ] On sites where wc-admin has not been disabled with the `woocommerce_admin_disabled` filter
- [ ] Revise the Low Stock link to point to `wp-admin/admin.php?page=wc-admin&type=lowstock&path=%2Fanalytics%2Fstock`
- [ ] Revise the out of stock link to `wp-admin/admin.php?page=wc-admin&type=outofstock&path=%2Fanalytics%2Fstock`

### PR DESCRIPTION
**Pull request by louwie17**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Updates the out of stock, low stock links in the WooCommerce widget in the WP dashboard. The point to the new analytics page now if WooCommerce admin is not disabled.

Closes #28570 

### How to test the changes in this Pull Request:

1.  Pull down this branch and make sure Woo is enabled
2. Navigate to the WP dashboard (**Dashboard > Home**) and scroll down to the **WooCommerce Status** widget
3. If no products it should show. **0 products** and **out of stock**, click on this
4. This should direct to `/wp-admin/wp-admin/admin.php?page=wc-admin&type=outofstock&path=%2Fanalytics%2Fstock` and have **Out of Stock** select in the dropdown.
5. Navigate back to the **Woocommerce Status** widget in the WP dashboard
6. This time click on the **low in stock** button
7. This should redirect you to the same analytics page, but this time have **Low Stock** selected in the dropdown.
8. Add `add_filter( 'woocommerce_admin_disabled', '__return_true' );` to your themes `functions.php` file.
9. Load the WP dashboard again
10. Both the above buttons should now be directing to the old **WooCommerce > Reports** screen (check both low stock and out of stock)

product-stock-links

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enhancement - Update the woo widget stock links to the new analytics page.

### CODE REVIEW

File: includes/admin/class-wc-admin-dashboard.php [hunk 1]
```diff
@@ -107,6 +107,8 @@ private function get_sales_report_data() {
 		public function status_widget() {
 			include_once dirname( __FILE__ ) . '/reports/class-wc-admin-report.php';
 
+			$is_wc_admin_disabled = apply_filters( 'woocommerce_admin_disabled', false );
```
Comment (louwie17) [inline includes/admin/class-wc-admin-dashboard.php:110]: Added here, as we will be updating the other links in other tickets.

File: includes/admin/class-wc-admin-dashboard.php [hunk 2]
```diff
@@ -246,6 +250,13 @@ private function status_widget_stock_rows() {
 
 			$transient_name   = 'wc_outofstock_count';
 			$outofstock_count = get_transient( $transient_name );
+			$lowstock_link    = 'admin.php?page=wc-reports&tab=stock&report=low_in_stock';
+			$outofstock_link  = 'admin.php?page=wc-reports&tab=stock&report=out_of_stock';
+
+			if ( false === $is_wc_admin_disabled ) {
+				$lowstock_link   = 'wp-admin/admin.php?page=wc-admin&type=lowstock&path=%2Fanalytics%2Fstock';
```
Comment (roykho) [inline includes/admin/class-wc-admin-dashboard.php:257]: The `wp-admin/` is not needed here as it's already prepended by `admin_url()` Same with the line below this.

Comment (louwie17) [reply includes/admin/class-wc-admin-dashboard.php]: Thanks, that's what I get from copying it straight from the ticket. It's been updated.

**Review by roykho — APPROVED** [review]
Thanks for the PR! LGTM. Approving this PR pending other party's review.

**Review by samueljseay — APPROVED** [review]
LGTM 🚀

**Pull request merged**